lib/slack_rewriter_web/controllers/event_api_controller.ex

```diff
@@ -0,0 +1,56 @@
+defmodule SlackRewriterWeb.EventApiController do
+  use SlackRewriterWeb, :controller
+
+  alias SlackRewriter.Application.UserRegistry
+  alias SlackRewriter.Bridge.Slack
+
+  alias SlackRewriter.Application.SlackRewriter
+
+  def event_handler(
+        conn = %{body_params: %{"challenge" => challenge, "token" => _token, "type" => _type}},
+        _params
+      ) do
+    # recall: verify slack token (better: read doc about signed secrets)
+    conn
+    |> put_resp_content_type("application/x-www-form-urlencoded")
+    |> json(%{challenge: challenge})
+  end
+
+  def event_handler(
+        conn = %{
+          body_params: %{
+            "event" => %{"type" => "message"}
```

Comment:
```suggestion
            "event" => %{"type" => "message"} = event
```
puoi fare contemporaneamente pattern match e "assegnazione", così ti risparmi un po' di passaggi sotto
```elixir
SlackRewriter.react_to_message(event)
```